src/main/java/org/pentaho/metadata/repository/IMetadataDomainRepositoryDatasourceType.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2022 Hitachi Vantara..  All rights reserved.
+ */
+
+package org.pentaho.metadata.repository;
+
+import java.util.Set;
+
+/**
+ * This interface defines a metadata domain repository, used to maintain a system-wide set of metadata domains.
+ * Includes the ability to retrieve Domains based on data source type.
+ */
+public interface IMetadataDomainRepositoryDatasourceType extends IMetadataDomainRepository {
```

Comment:
Am I correct to conclude that the types of data sources are not fixed/closed, or are established elsewhere, and that thus it would not make sense to add these methods to the base interface?

Otherwise, would it be possible to add these to the base interface with `default` implementations?

Reply:
The concept of Metadata and Datasource wizards aren't really established in pentaho/pentaho-metadata until pentaho/data-access. 

But I do see your point, we could avoid creating a whole new interface by providing a default implementation in IMetadataDomainRepository.java. I guess the default implementation for getting metadata specific domain ids would be to return empty set?


More detail:

the dependency is via repos:
pentaho/pentaho-platform depends on pentaho/pentaho-metadata (interfaces)
pentaho/data-access depends on  pentaho/pentaho-metadata (interfaces) and pentaho/pentaho-platform (implementation)

Currently, the concept of data source wizard isn't introduced until pentaho/data-access like we have https://github.com/pentaho/data-access/blob/master/core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/DataSourceWizardService.java which is kind of the problem that I'm trying to correct, introducing the concepts of metadata/datasourcewizard earlier in the pentaho/pentaho-platform code.

There are other pieces of code that utilized the original interface IMetadataDomainRepository see the bullet #2 in the description of https://jira.pentaho.com/browse/BACKLOG-36079. Those classes don't need to different between metadata and datasource wizard. For example: https://github.com/pentaho/mql-editor/blob/master/impl/src/main/java/org/pentaho/commons/metadata/mqleditor/editor/service/util/MQLEditorServiceDelegate.java#L96-L102